### PR TITLE
#14382: fix invalid lengths in prefetcher test

### PR DESCRIPTION
**Pull request by nhuang-tt**
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/14382)

### Problem description
When the prefetcher test is run on Wormhole B0 and Blackhole the assertions in gen_dram_packed_read_cmd are being hit due to invalid lengths being provided.

The tests affected are the smoke test and packed read.

### What's changed
- Some read lengths being made by calculations are not aligned correctly. Update them to be aligned to the DRAM alignment
- Limit the read lengths to between max and min read size

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_prefetcher.cpp [hunk 1]
```diff
@@ -461,15 +461,15 @@ void gen_dram_packed_read_cmd(Device *device,
...
+    for (const auto& length : lengths) {
+        total_length += length;
     }
     gen_bare_dispatcher_unicast_write_cmd(device, dispatch_cmds, worker_core, device_data, total_length);
     add_prefetcher_cmd(prefetch_cmds, cmd_sizes, CQ_PREFETCH_CMD_RELAY_INLINE_NOFLUSH, dispatch_cmds);
 
     uint32_t page_size = 1 << log_page_size;
     int count = 0;
-    for (auto length : lengths) {
+    for (const auto& length : lengths) {
```
Comment (pgkeller) [inline tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_prefetcher.cpp:472]: why use a const ref when we just want a value (copying an int has no cost here)

File: tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_prefetcher.cpp [hunk 2]
```diff
@@ -880,18 +880,19 @@ void gen_packed_read_test(Device *device,
                           vector<uint32_t>& prefetch_cmds,
                           vector<uint32_t>& cmd_sizes,
                           DeviceData& device_data) {
-
+    static constexpr uint32_t min_read_size = 128;
```
Comment (pgkeller) [inline tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_prefetcher.cpp:883]: where does 128 come from?

**Review by pgkeller — CHANGES_REQUESTED** [review]

File: tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_prefetcher.cpp (hunk 2 above)
Comment (nhuang-tt) [reply tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_prefetcher.cpp:883]: 128 was from the original code. If the length was less than 128 then it set the length to 128. I extracted it out to increase readability.

File: tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_prefetcher.cpp (hunk 1 above)
Comment (nhuang-tt) [reply tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_prefetcher.cpp:472]: Ack

**Review by pgkeller — APPROVED** [review]

**Pull request merged**